src/classes/Shapes/MeshShape.js

```diff
@@ -5,56 +5,61 @@
  * @constructor
  */
 Goblin.MeshShape = function( vertices, faces, material ) {
-	this.vertices = vertices;
-
-	this.triangles = [];
-	for ( var i = 0; i < faces.length; i += 3 ) {
-		this.triangles.push( new Goblin.TriangleShape( vertices[faces[i]], vertices[faces[i+1]], vertices[faces[i+2]] ) );
-	}
-
-	/**
-	 * the convex mesh's volume
-	 * @property volume
-	 * @type {number}
-	 */
-	this.volume = 0;
-
-	/**
-	 * coordinates of the mesh's COM
-	 * @property center_of_mass
-	 * @type {vec3}
-	 */
-	this.center_of_mass = new Goblin.Vector3();
-
-	/**
-	 * used in computing the mesh's center of mass & volume
-	 * @property _intergral
-	 * @type {Float32Array}
-	 * @private
-	 */
-	this._integral = new Float32Array( 10 );
-
-	this.hierarchy = new Goblin.BVH( this.triangles ).tree;
-
-	var polygon_faces = this.triangles.map(
-		function( triangle ) {
-			return new Goblin.GjkEpa.Face(
-				null,
-				{ point: triangle.a },
-				{ point: triangle.b },
-				{ point: triangle.c }
-			);
-		}
-	);
-
-	Goblin.ConvexShape.prototype.computeVolume.call( this, polygon_faces );
-
-	this.aabb = new Goblin.AABB();
-	this.calculateLocalAABB( this.aabb );
-
-	this.material = material || null;
-
-	this.shape = Goblin.MeshShape;
+    this.shapeType = Goblin.Shapes.Type.MeshShape;
```

Comment:
Let's put a sticky note here begging for rethinking and optimizing mesh shape. Looks like we spend a lot of time on quite terrible things like copying points around.

Reply:
I've created a ticket EN-272 to deal with it